pkg/apis/certmanager/v1alpha1/types.go

```diff
@@ -74,8 +74,23 @@ type IssuerSpec struct {
 }
 
 type IssuerConfig struct {
-	ACME *ACMEIssuer `json:"acme,omitempty"`
-	CA   *CAIssuer   `json:"ca,omitempty"`
+	ACME  *ACMEIssuer  `json:"acme,omitempty"`
+	CA    *CAIssuer    `json:"ca,omitempty"`
+	Vault *VaultIssuer `json:"vault,omitempty""`
+}
+
+type VaultIssuer struct {
+	// Vault authentication
+	Auth VaultAuth `json:"auth"`
+	// Server is the vault connection address
+	Server string `json:"server"`
+	// Vault URL path to the certificate role
+	Path string `json:"path"`
```

Comment:
This implies a 1:1 binding between Issuer and a *role* within a PKI backend. I think this makes sense to be honest, and I like keeping it simple here, but wanted to highlight it.

Reply:
Perhaps before 1.0 we could add some sort of 'role binding' support. This would probably be a breaking change... we'll see what users want.